environment/src/main/java/com/sequenceiq/environment/parameters/validation/validators/parameter/AwsParameterValidator.java

```diff
@@ -54,17 +54,21 @@ public ValidationResult validate(EnvironmentValidationDto environmentValidationD
             LOGGER.debug("No aws parameters defined.");
             return validationResultBuilder.build();
         }
-        if (StringUtils.isNotBlank(awsParametersDto.getS3GuardTableName())) {
-            LOGGER.debug("S3Guard table name defined: {}", awsParametersDto.getS3GuardTableName());
-            boolean tableAlreadyAttached = isTableAlreadyAttached(environmentDto, awsParametersDto);
-            if (tableAlreadyAttached) {
-                validationResultBuilder.error(String.format("S3Guard Dynamo table '%s' is already attached to another active environment. "
-                        + "Please select another unattached table or specify a non-existing name to create it. "
-                        + "Refer to Cloudera documentation at %s for the required setup.",
-                        awsParametersDto.getS3GuardTableName(),
-                        DocumentationLinkProvider.awsDynamoDbSetupLink()));
-            } else {
-                determineAwsParameters(environmentDto, parametersDto);
+        //This flag is introduced to skip S3Guard Table validation in case of UpdateAwsDiskEncryptionParameters call.
+        boolean validateOnlyAwsEncryptionParameters = validateOnlyAwsEncryptionParameters(environmentDto);
+        if (Objects.isNull(awsParametersDto.getAwsDiskEncryptionParametersDto()) &&  !validateOnlyAwsEncryptionParameters) {
```

Comment:
Sorry, I do not get this condition. If there is no disk encryption in the environment, `getAwsDiskEncryptionParametersDto()` will be `null` (I suppose), so `validateOnlyAwsEncryptionParameters` will be `true`. In that case this condition will never be satisfied, so the S3 Guard table will not be verified. Is that what we want?

Reply:
The problem I am trying to address here is : In environment which already have a s3guard table name setup in it, no encryption parameters available(as we do not support updating the existing CMK but only support adding a new CMK if it does not already exist in an environment) and then user runs a update-aws-disk-encryption-parameters cli with a new CMK, we should not validate the s3guard table name(as it leads to failure as the env already have a s3guard and it tries to update that again , but the s3 guard table name is same it ends up in an error ```
S3Guard Dynamo table '%s' is already attached to another active environment. "
                                    + "Please select another unattached table or specify a non-existing name to create it. "
                                    + "Refer to Cloudera documentation at %s for the required setup.
```

To avoid this behaviour we should not validate s3 guard in case of updating the CMK for an existing environment.

Reply:
@lajosrodek Gentle remainder to review this :-)

Reply:
My bad. I mistakenly assumed that `awsParametersDto` was coming from the environment details. But it actually is the edit change requested, if I am right. Okay, this then looks fine. :smile: